pkg/router/template/plugin.go

```diff
@@ -265,9 +265,27 @@ func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool, lookupSvc
 		wasIdled = true
 	}
 
-	out := make([]Endpoint, 0, len(endpoints.Subsets)*4)
+	// Count the number of endpoint addresses (we have to do this first since we need the result when we loop below)
+	// But the two loop conditions must be the same
+	numEndpoints := 0
+	for _, s := range subsets {
+		for _, p := range s.Ports {
+			if excludeUDP && p.Protocol == kapi.ProtocolUDP {
+				continue
+			}
+
+			numEndpoints += len(s.Addresses)
+		}
+	}
+
+	// We want to skip health checking when we are idled or when there is only one address
+	noHealthCheck := false
+	if wasIdled || numEndpoints == 1 {
+		noHealthCheck = true
+	}
 
-	// TODO: review me for sanity
+	// Now build the actual endpoints we pass to the template
+	out := make([]Endpoint, 0, numEndpoints)
```

Comment:
I would just do this and if there is more than 1 end point (after line 325), loop over endpoints setting the flag. The argument to do this suggests that most of the time the count will be 1. Looping to find that out seems inefficient.